src/lib.rs

```diff
@@ -212,18 +283,29 @@ pub struct Form {
 }
 
 impl Client {
-    /// Create a new `Client` associated with a new WebDriver session on the server at the given
-    /// URL.
+    /// Create a new [`Client`][crate::Client] associated with a chosen HttpsConnector and a new WebDriver session on the server
```

Comment:
You should not need to give `[crate::Client]` here — rustdoc will figure that automatically on recent compiler versions.

Also, in Rust, the first line of documentation for a function is usually a very short summary of what the function does, with subsequent paragraphs going into more detail. I'd suggest:
```rust
/// Connect to the WebDriver host running the given address.
///
/// The provided `connector` is used to establish the connection to the WebDriver host, and should generally be one that supports HTTPS, as that is commonly required by WebDriver implementations.
```

(though line-wrapped appropriately)